spec/formatting.md

```diff
@@ -19,6 +19,112 @@ the local variable takes precedence.
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
 
+## Pattern Selection
+
+When formatting a message with one or more selectors,
+the Pattern of one of Variants must be selected for formatting.
+
+An implementation may use any pattern selection method,
+as long as its observable behaviour matches the results of the method defined here.
+
+### Setup
+
+To select a Pattern,
+first the resolved values _res_ of the Selector must be determined:
+
+1. Let _res_ be a new empty list of resolved values that support selection.
+2. For each Expression _exp_ of the message's Selector Expressions,
+   1. Let _rv_ be the resolved value of _exp_.
+   2. If selection is supported for _rv_:
+      1. Append _rv_ as the last element of the list _res_.
+   3. Else:
+      1. Emit a Selection Error.
+      2. Let _nomatch_ be a resolved value for which selection always fails.
+      3. Append _nomatch_ as the last element of the list _res_.
+
+The shape of the resolved values must be determined by each implementation,
```

Comment:
This is trying to be explicit about the parts of the selection process that are left for the implementation, and makes it clear that the spec explicitly leaves out the shape of any resolved values, or how to work with them.

My personal preference would be for the spec to be built in terms of explicit, well-defined resolved values, but @mihnita in particular has strongly pushed back on this. Unfortunately, this means that it's tricky to talk about e.g. what the value of `$num` is here:

```
let $num = {(1) :number minimumFractionDigits=1}
match {$num}
when one {You have {$num} thing}
when * {You have {$num} things}
```

I would find it much easier to say that `$num` resolves to be an instance of a MessageNumber with a value `1` and an options bag `{ minimumFractionDigits: 1 }`. When you use a MessageNumber as a selector, _this_ is how the matching works, and when you use it in a placeholder, _that_ is what you get when formatting to a string, and _other_ is the behaviour when formatting to parts.

But taking considerations of realpolitik into account, we appear to need to define the spec without any such MessageNumber constructs, and hence end up with this circumlocution around resolved values.

Reply:
> But taking considerations of realpolitik into account...

Hmm, we should continue to stay focused on our objective technical arguments, and of course doing our usual due diligence around them (multiple alternatives, pros & cons, eval criteria for a preference). I'm not sure that we're all so far apart in our thinking on this topic, so giving into non-technical concerns slows us down from achieving our best work.

I would like to get a clearer picture on the previous discussion from @mihnita. It could be here, but also could be in a meeting. Maybe after hearing more details, it would help us identify the precise technical sticking points? And hopefully my response to the [other comment](https://github.com/unicode-org/message-format-wg/pull/333/files#r1095475687) helps add some context to the topic by identifying how we might be talking past each other.